src/config/rcxml.c

```diff
@@ -1042,10 +1047,17 @@ post_processing(void)
 		struct libinput_category *l = libinput_category_create();
 		assert(l && libinput_category_get_default() == l);
 	}
-	if (!wl_list_length(&rc.workspace_config.workspaces)) {
-		struct workspace *workspace = znew(*workspace);
-		workspace->name = xstrdup("Default");
-		wl_list_append(&rc.workspace_config.workspaces, &workspace->link);
+
+	int workspaces_configured = wl_list_length(&rc.workspace_config.workspaces);
+	if (workspaces_configured < rc.workspace_config.count) {
+		struct workspace *workspace;
+		char workspace_name[32]; // Maximum length of workspace name "Workspace X"
```

Comment:
I’d remove comment because it should be self-explanatory. Also, we ought to allow for translation in future in which case 32 might not be enough. But have to sort translation as a different patch.

We try to use /* */ comments instead of //.
In fact I’m surprised that the checkpatch.pl script didn’t catchy that.

Reply:
> I’d remove comment because it should be self-explanatory. Also, we ought to allow for translation in future in which case 32 might not be enough. But have to sort translation as a different patch.
> 
> We try to use /* */ comments instead of //. In fact I’m surprised that the checkpatch.pl script didn’t catchy that.

Do you want me to dynamically declare this? For example, using a linked list may help?

Reply:
I could try something like this maybe:
Replace lines 1054 to 1056 with
```c
        char *workspace_name = malloc(sizeof(char) * 32); /* Dynamically allocate memory */
        snprintf(workspace_name, 32, "Workspace %d", i + 1);
        workspace->name = workspace_name;
```

Would love to hear your thoughts!

Reply:
This should fix the fact that
> in which case 32 might not be enough.

Reply:
It could be done this way to make it easier to adjust the size later on when we add the translation. I am not sure if we should do that within this PR though but maybe rather change it when adding the translation. And then also use `xmalloc()` instead of `malloc()` (our wrapper for it, see `include/common/mem.h`).

Reply:
Will revert back with the necessary changes.

Reply:
> I could try something like this maybe:
> 
> ```c
>         char *workspace_name = malloc(sizeof(char) * 32); /* Dynamically allocate memory */
>         snprintf(workspace_name, 32, "Workspace %d", i + 1);
>         workspace->name = workspace_name;
> ```
> 
> Would love to hear your thoughts!

What you'd normally do here is to use a string library of some sort :smile:
There is glib's `g_strdup_printf()` and with `DEFINE_GNU_SOURCE` you get various heap-alloc varieties of sprintf family such as `asprintf()`. We could just use `asprintf()` but not everyone likes using compilers / C-libraries that support these so it always stirs up debate which we try to stay clear of.

The problem with the code snippet above it that you're still defining the size at 32.

Don't worry about it for now. There are other areas of the code-base that would benefit from this anyway. The simplest way is to do a wrapper around `vsnprintf()` to first get the size, then allocate the buffer, then populate it.
I'll do a PR :smile: